broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java

```diff
@@ -1960,11 +1960,20 @@ public RemotingCommand deleteExpiredCommitLog() {
     }
 
     public RemotingCommand cleanUnusedTopic() {
-        LOGGER.warn("invoke cleanUnusedTopic start.");
+        LOGGER.info("AdminBrokerProcessor#cleanUnusedTopic: start.");
         final RemotingCommand response = RemotingCommand.createResponseCommand(null);
-        brokerController.getMessageStore().cleanUnusedTopic(brokerController.getTopicConfigManager().getTopicConfigTable().keySet());
-        LOGGER.warn("invoke cleanUnusedTopic end.");
-        response.setCode(ResponseCode.SUCCESS);
+        Set<String> topicManagerTopics = Sets.newHashSet(
+            brokerController.getTopicConfigManager().getTopicConfigTable().keySet());
+        int responseCode = ResponseCode.SUCCESS;
+        if (this.brokerController.getMessageStore() instanceof DefaultMessageStore) {
```

Comment:
Why should we add this restriction？ IMO, there is no need to care about the underlying implementation outside the store module.

Reply:
I guess only the implementation class DefaultMessageStore can obtain consumeQueueTable

Reply:
Then we maybe consider exposing the interface to get all cq or sinking the implementation to the store module.

Reply:
I already change it.